content/en/tracing/trace_collection/library_config/nodejs.md

```diff
@@ -26,87 +26,211 @@ Tracer settings can be configured with the following environment variables:
 
 `DD_TRACE_AGENT_HOSTNAME`
-: **Default**: `localhost`<br>
+: **Configuration**: `hostname`<br>
+**Default**: `localhost`<br>
 The address of the Agent that the tracer submits to.
 
 `DD_TRACE_AGENT_PORT`
-: **Default**: `8126`<br>
+: **Configuration**: `port`<br>
+**Default**: `8126`<br>
 The port of the Trace Agent that the tracer submits to.
 
 `DD_DOGSTATSD_PORT`
-: **Default**: `8125`<br>
+: **Configuration**: `dogstatsd.port`<br>
+**Default**: `8125`<br>
 The port of the DogStatsD Agent that metrics are submitted to.
 
 `DD_LOGS_INJECTION`
-: **Default**: `false`<br>
+: **Configuration**: `logInjection`<br>
+**Default**: `false`<br>
 Enable automatic injection of trace IDs in logs for supported logging libraries.
 
 `DD_TRACE_SAMPLE_RATE`
-: Percentage of spans to sample as a float between `0` and `1`. Defaults to the rates returned by the Datadog Agent.
+: **Configuration**: `sampleRate`<br>
+**Default**: Defaults to deferring the decision to the agent.<br>
+Controls the ingestion sample rate (between 0.0 and 1.0) between the agent and the backend.
 
 `DD_TRACE_RATE_LIMIT`
-: Percentage of spans to sample as a float between `0` and `1`. Defaults to `100` when `DD_TRACE_SAMPLE_RATE` is set. Otherwise, delegates rate limiting to the Datadog Agent.
+: **Configuration**: `rateLimit`<br>
+**Default**: Defaults to `1.0` when `DD_TRACE_SAMPLE_RATE` is set. Otherwise, delegates rate limiting to the Datadog Agent.
+Ratio of spans to sample as a float between `0.0` and `1.0`. <br>
 
-`DD_TRACE_SAMPLING_RULES` 
-: A JSON array of objects. Each object must have a "sample_rate", and the "name" and "service" fields are optional. The "sample_rate" value must be between 0.0 and 1.0 (inclusive). Rules are applied in configured order to determine the trace's sample rate. If omitted, the tracer defers to the Agent to dynamically adjust sample rate across all traces.
+`DD_TRACE_SAMPLING_RULES`
+: **Configuration**: `samplingRules`<br>
+**Default**: `[]`<br>
+Sampling rules to apply to priority sampling. A JSON array of objects. Each object must have a `sample_rate` value between 0.0 and 1.0 (inclusive). Each rule has an optional `name` and `service` fields, which are regexes to match against a trace's `service` and `name`. Rules are applied in configured order to determine the trace's sample rate. If omitted, the tracer defers to the Agent to dynamically adjust sample rate across all traces.
 
 `DD_SPAN_SAMPLING_RULES`
-: **Default**: `undefined`<br>
-A JSON array of objects. Rules are applied in configured order to determine the span's sample rate. The `sample_rate` value must be between 0.0 and 1.0 (inclusive).
+: **Configuration**: `spanSamplingRules`<br>
+**Default**: `[]`<br>
+Span sampling rules for ingesting single spans, in cases where the whole trace is dropped. A JSON array of objects. Rules are applied in configured order to determine the span's sample rate. The `sample_rate` value must be between 0.0 and 1.0 (inclusive).
```

Comment:
Maybe clarify that it is to _keep_ certain spans despite the trace being dropped? Reads a little weird to me like you can sample individual spans but then it will get dropped anyway for some reason. 🤔

Reply:
I don't have a clue how it works, just copypasta from existing sources. How would you like it rewritten?

Reply:
Something like this, maybe?

```suggestion
Span sampling rules to keep individual spans when the rest of the trace would otherwise be dropped. A JSON array of objects. Rules are applied in configured order to determine the span's sample rate. The `sample_rate` value must be between 0.0 and 1.0 (inclusive).
```